frontend/src/pages/subcomponents/TextCheck.tsx

```diff
@@ -1,45 +1,57 @@
-import {Button, Grid} from "@mui/material";
+import {
+    Box,
+    Grid
+} from "@mui/material";
 import {TextResponse, WordAvail} from "../../services/model";
+import WordDropdown from "./WordDropdown";
 
 interface TextCheckProps {
-    splitText: TextResponse
+    splitText: TextResponse,
+    ids: string[],
+    setIds: (ids: string[]) => void
 }
 
 export default function TextCheck(props: TextCheckProps) {
 
-    function getWordButton(word: WordAvail) {
+    function generateIdSetter(index: number){
+        return (id: string) => {
+            let localIds = props.ids;
+            localIds[index] = id;
+            props.setIds(localIds);
+        }
+    }
+
+    function getWordButton(word: WordAvail, index: number) {
         let myColor: string = "#fff";
-        let myTextDecoration: string = "none";
+        // let myTextDecoration: string = "none";
 
         if (word.availability === "PUBLIC") {
             myColor = "#12670c";
         } else if (word.availability === "INVALID") {
             myColor = "#881111";
-            myTextDecoration = "line-through";
+            // myTextDecoration = "line-through";
```

Comment:
Are those still needed?

Reply:
i would like to display it like this but i dont really know how to do it with the new structure and its not important. i will just leave that comment as a reminder